src/main/java/com/example/taxiservicespring/service/impl/TripServiceImpl.java

```diff
@@ -0,0 +1,239 @@
+import com.example.taxiservicespring.service.repository.LocationRepository;
+import com.example.taxiservicespring.service.repository.TripRepository;
+import com.example.taxiservicespring.util.DistanceCalculator;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class TripServiceImpl implements TripService {
+    private static final String DATE_FORMAT = "dd.MM.yy";
+    private static final BigDecimal MIN_DICTANCE = BigDecimal.ONE;
+    private static final BigDecimal AVG_SPEED = BigDecimal.valueOf(0.3); // car average speed in km/min
+    private static final int SCALE = 2;
+    @Value("#{${discount}}")
+    private final Map<BigDecimal, BigDecimal> discounts;
+    private final TripRepository tripRepository;
+    private final LocationRepository locationRepository;
+    private final CategoryRepository categoryRepository;
+    private final CarRepository carRepository;
+
+    @Override
+    public TripDto find(long id) {
+        log.info("get trip by id {}", id);
+        Trip trip = tripRepository.find(id);
+        return TripMapper.INSTANCE.mapTripDto(trip);
+    }
+
+    @Override
+    public List<CarDto> findCarsByTripId(long tripId) {
+        List<Car> cars = tripRepository.findCarsByTripId(tripId);
+        return cars.stream()
+                .map(car -> CarMapper.INSTANCE.mapCarDto(car))
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    public TripConfirmDto create(TripCreateDto tripCreateDto) {
+        log.info("create new trip");
+        Category category = categoryRepository.find(tripCreateDto.getCategoryId());
+        List<CarDto> cars = new ArrayList<>();
+
+        if (tripCreateDto.isMultipleCars()) {
+            cars = carRepository.findCars(tripCreateDto.getCategoryId(), tripCreateDto.getCapacity())
+                    .stream()
+                    .map(car -> CarMapper.INSTANCE.mapCarDto(car))
+                    .collect(Collectors.toList());
+            } else if (tripCreateDto.isIgnoreCategory()) {
+            Car car = carRepository.findByCapacity(tripCreateDto.getCapacity());
+            category = categoryRepository.find(car.getCategoryId());
+            cars.add(CarMapper.INSTANCE.mapCarDto(car));
+        } else {
+            Car car = carRepository.find(tripCreateDto.getCategoryId(), tripCreateDto.getCapacity());
+            cars.add(CarMapper.INSTANCE.mapCarDto(car));
+        }
+        BigDecimal distance = getDistanceForTrip(tripCreateDto.getOriginId(), tripCreateDto.getDestinationId());
+        BigDecimal price = getPrice(category.getPrice(), distance).multiply(BigDecimal.valueOf(cars.size()));
+        BigDecimal discount = getDiscount(tripCreateDto.getPersonId(), price);
+        BigDecimal total = price.subtract(discount);
+        LocalTime waitTime = getWaitTime(tripCreateDto.getOriginId(), cars);
+        return TripConfirmDto.builder()
```

Comment:
I would recommend to put the code which builds the TripConfirmDto to separate method called createTripConfirmDto() and use it instead of this. It will increase readability